tests/e2e/customsuites/multivolume.go

```diff
@@ -0,0 +1,324 @@
+	//
+	// Validates that a single pod can mount multiple S3 volumes
+	// and access them independently without cross-contamination.
+	//
+	// This test verifies:
+	// - Multiple S3 volume mounts within a single pod
+	// - Volume isolation (data in one volume is not visible in another)
+	// - Proper mount path handling for multiple volumes
+	ginkgo.It("should allow a pod to access multiple volumes", func(ctx context.Context) {
+		// Create first volume resource with mount options
+		resource1 := createVolumeResourceWithMountOptions(ctx, l.config, pattern, mountOptions)
+		l.resources = append(l.resources, resource1)
+
+		// Create second volume resource with mount options
+		resource2 := createVolumeResourceWithMountOptions(ctx, l.config, pattern, mountOptions)
+		l.resources = append(l.resources, resource2)
+
+		// Create pod with both volumes
+		ginkgo.By("Creating pod with multiple volumes")
+		pod := e2epod.MakePod(f.Namespace.Name, nil, []*v1.PersistentVolumeClaim{resource1.Pvc, resource2.Pvc}, admissionapi.LevelRestricted, "")
+		pod.Name = fmt.Sprintf("multipod-%s", uuid.New().String()[:8])
+		podModifierNonRoot(pod)
+
+		// Modify the pod to have two distinct mount paths for the volumes
+		pod.Spec.Containers[0].VolumeMounts[0].MountPath = "/mnt/volume1"
+		pod.Spec.Containers[0].VolumeMounts[1].MountPath = "/mnt/volume2"
+
+		pod, err := createPod(ctx, f.ClientSet, f.Namespace.Name, pod)
+		framework.ExpectNoError(err)
+
+		defer func() {
+			framework.ExpectNoError(e2epod.DeletePodWithWait(ctx, f.ClientSet, pod))
+		}()
+
+		// Write and read data from both volumes
+		ginkgo.By("Writing data to first volume")
+		vol1Path := "/mnt/volume1"
+		fileInVol1 := fmt.Sprintf("%s/file1.txt", vol1Path)
+		seed1 := time.Now().UTC().UnixNano()
+		toWrite := 1024 // 1KB
+
+		checkWriteToPath(f, pod, fileInVol1, toWrite, seed1)
+		checkReadFromPath(f, pod, fileInVol1, toWrite, seed1)
+
+		ginkgo.By("Writing data to second volume")
+		vol2Path := "/mnt/volume2"
+		fileInVol2 := fmt.Sprintf("%s/file2.txt", vol2Path)
+		seed2 := time.Now().UTC().UnixNano() + 1 // Different seed
+
+		checkWriteToPath(f, pod, fileInVol2, toWrite, seed2)
+		checkReadFromPath(f, pod, fileInVol2, toWrite, seed2)
+
+		// Verify the volumes are indeed separate by checking file existence
+		ginkgo.By("Verifying volume separation")
+		_, stderr, err := e2evolume.PodExec(f, pod, fmt.Sprintf("ls %s", fmt.Sprintf("%s/file2.txt", vol1Path)))
+		gomega.Expect(err).To(gomega.HaveOccurred())
+		gomega.Expect(stderr).To(gomega.ContainSubstring("No such file or directory"))
+
+		_, stderr, err = e2evolume.PodExec(f, pod, fmt.Sprintf("ls %s", fmt.Sprintf("%s/file1.txt", vol2Path)))
+		gomega.Expect(err).To(gomega.HaveOccurred())
+		gomega.Expect(stderr).To(gomega.ContainSubstring("No such file or directory"))
+	})
+
+	// Test 3: Data persistence across pod recreations
+	// This test validates the following configuration:
+	//
+	//  [Pod-1 (writes)]  →  ✓ Delete  →  [Pod-2 (reads)]
```

Comment:
So that's on me, but when I started reading the description, I thought you meant delete a file. I quickly realized that no. So do what you want with this suggestion 😅  
```suggestion
	//  [Pod-1 (writes)]  →  ✓ Delete Pod-1 →  [Pod-2 (reads)]
```

Reply:
I will take it, while reviewing my code, I felt like that as well :)

Reply:
you found the missing piece to this comment, it did feel off to me as well, you were right to feel that way